### PR TITLE
[emitter-python] update DPG how-to guide

### PR DESCRIPTION
**Pull request by msyyc**
update how-to guide for python part

### CODE REVIEW

File: docs/howtos/DataPlane Generation - DPG/03convenient.mdx [hunk 1]
```diff
@@ -168,9 +167,9 @@ The two possible value for the `Access` enum are `internal` and `public`.
 <TabItem value="python" label="Python">
 
 ```python
-class _GetModel:
-  ...
-# FIXME
+# can't import form models directly
+from petstorenamespace.models import GetModel # will report error
+from petstorenamespace.models._models import GetModel # OK but not recommended in Python
```
Comment (iscai-msft) [inline docs/howtos/DataPlane Generation - DPG/03convenient.mdx:172]: we shouldn't document private imports

**Comment by msyyc** [comment]
already done https://github.com/Azure/typespec-azure/blob/main/docs/howtos/Client%20Generation/03convenient.mdx

File: docs/howtos/DataPlane Generation - DPG/03convenient.mdx (hunk 1 above)
Comment (iscai-msft) [reply docs/howtos/DataPlane Generation - DPG/03convenient.mdx:172]: I think we should just not even include the private imports

Comment (msyyc) [reply docs/howtos/DataPlane Generation - DPG/03convenient.mdx]: Updated @iscai-msft

File: docs/howtos/Client Generation/05union.mdx [hunk 2]
```diff
@@ -37,11 +37,11 @@ op completions(@body input: CompletionInput): CompletionResult;
 <TabItem value="python" label="Python">
 
 ```python
-def analyze(*, mode: Literal["strict", "lenient"], image: bytes) -> AnalyzeResult:
+def analyze(image: bytes, *, mode: Literal["strict", "lenient"]) -> AnalyzeResult:
   ...
 
 class CompletionInput:
-  input: str | List[str] = rest_field(readonly=True)
+  input: Union[str, List[str]] = rest_field(readonly=True)
```
Comment (msyyc) [inline docs/howtos/Client Generation/05union.mdx:44]: Introduced in 3.10: https://docs.python.org/3/library/stdtypes.html#types-union and 3.8/3.9 doesn't support yet.

**Review by iscai-msft — APPROVED** [review]

**Pull request merged**